core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -1556,6 +1681,133 @@ public void warmUp() {
         }
     }
 
+    private void checkDetachedMetadata(long timestamp) {
+        // load/check _meta
+        other2.of(detachedPath).concat(META_FILE_NAME).$();
+        if (!ff.exists(other2)) {
+            // Backups and older versions of detached partitions will not have _meta
+            LOG.info().$("detached _meta not found, skipping check [path=").$(other2).I$();
+            return;
+        }
+        try {
+            detachedMetadata.deferredInit(other2, ColumnType.VERSION);
+            if (metadata.getId() != detachedMetadata.getId()) {
+                throw CairoException.detachedMetadataMismatch("table_id");
+            }
+            // check column name, type and isIndexed
+            for (int colIdx = 0; colIdx < columnCount; colIdx++) {
+                String columnName = metadata.getColumnName(colIdx);
+                int detColIdx = detachedMetadata.getColumnIndexQuiet(columnName);
+                if (detColIdx == -1) {
+                    throw CairoException.detachedMetadataMismatch("missing_column: " + columnName);
+                }
+                if (metadata.getColumnType(colIdx) != detachedMetadata.getColumnType(detColIdx)) {
```

Comment:
Deleted column will have negative type. They can mismatch but it just means that column is deleted.

Reply:
thank you!